src/main/java/interface_adapter/graph/GraphState.java

```diff
@@ -26,22 +26,43 @@ public void setGraphT(ListenableGraph<String, DefaultEdge> graphT) {
         this.graphT = graphT;
     }
 
+//    /**
+//     * Add a new node to the graph.
+//     * @param nodeName name of the new node
+//     */
+//    public void addNewNode(String nodeName) {
+//        graphT.addVertex(nodeName);
+//    }
+
     /**
-     * Add a new node to the graph.
-     * @param nodeName name of the new node
+     * Add a new node to the graph and connect it to its parent nodes.
+     * @param nodeName name of the existing node
      */
     public void addNewNode(String nodeName) {
         graphT.addVertex(nodeName);
+        for(String parentName : getAddedParents(nodeName)) {
+            graphT.addEdge(parentName, nodeName);
+        }
     }
 
+
     /**
-     * Add a new node to the graph and connect it to an exising node.
-     * @param nodeName1 name of the existing node
-     * @param nodeName2 name of the new node
+     * Return all ancestors of the current node that are added to the graph
+     * @param nodeName name of the node
+     * @return title of ancestors that are added to the graph
      */
-    public void addNewNode(String nodeName1, String nodeName2) {
-        graphT.addVertex(nodeName2);
-        graphT.addEdge(nodeName1, nodeName2);
+    private List<String> getAddedParents(String nodeName) {
+        Node currentNode = dao.getWikiHistory().getNode(nodeName);
```

Comment:
I would move this processing to the AddInteractor as it is accessing the InMemoryDataAccessObject without an interface and also creates a dependency which should be injected in the appBuilder that way the GraphState uses the persistent graph constructed at startup instead of having access to the InMemoryDataAccessObject